main/util/src/mill/util/Jvm.scala

```diff
@@ -617,12 +617,19 @@ object Jvm {
       id: String,
       ctx: Option[mill.define.TaskCtx.Log] = None,
       coursierCacheCustomizer: Option[FileCache[Task] => FileCache[Task]] = None,
-      jvmIndexVersion: String = "latest.release"
+      jvmIndexVersion: String = "latest.release",
+      useShortPaths: Boolean = false
   ): Result[os.Path] = {
     val coursierCache0 = coursierCache(ctx, coursierCacheCustomizer)
+    val shortPathDirOpt = Option.when(useShortPaths) {
+      if (isWin) os.Path(System.getenv("UserProfile")) / ".mill/cache/jvm"
+      else os.home / ".cache/mill/jvm"
```

Comment:
This snippet here looks a bit unusual. I thought we just use `os.home` for most things? Or are we using `UserProfile` in other parts of the code as well?

Reply:
Yes, `mill.bat` uses it [here](https://github.com/com-lihaoyi/mill/blob/58ab3d68b33b7ddaf6b63165bf5a08571494923b/mill.bat#L85). I also [used it](https://github.com/com-lihaoyi/mill/blob/58ab3d68b33b7ddaf6b63165bf5a08571494923b/runner/client/src/mill/runner/client/JLineNativeLoader.java#L41) in `JLineNativeLoader`, added in https://github.com/com-lihaoyi/mill/pull/4882.

Ideally, that directory should be obtained by a native Windows call rather than via the environment. Coursier does it via [directories-jvm](https://github.com/coursier/directories-jvm), but it's quite cumbersome (requires JNI or newer JVM foreign API stuff, or running a powershell script…). Given `mill.bat` relies on the env for that, it's better just to do the same in Mill IMO.